templates/model/spec.go

```diff
@@ -215,10 +225,9 @@ func (s *Step) Validate() error {
 		notZero(s.Pos, s.Desc, "desc"),
 		validateUnlessNil(s.Print),
 		validateUnlessNil(s.Include),
-		// TODO: add more action types:
-		// validateIfNotNil(s.RegexReplace),
-		// validateIfNotNil(s.StringReplace),
-		// validateIfNotNil(s.GoTemplate),
+		validateUnlessNil(s.RegexReplace),
+		validateUnlessNil(s.StringReplace),
+		validateUnlessNil(s.GoTemplate),
```

Comment:
Is it worth validating only one of action can be specified? Or it isn't?

Reply:
That's guaranteed by Spec.UnmarshalYAML(), take a look there. We choose a single field to parse into, then call Decode() just once, so only one of them can be set.

Reply:
You mean `Step.UnmarshalYAML()`?  I see it now.